src/integration-test/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitIntegrationTest.java

```diff
@@ -1189,6 +1195,124 @@ public void commit_NonConflictingDeletesGivenForExisting_ShouldCommitBoth()
     assertThat(another.get(gets.get(account4)).isPresent()).isFalse();
   }
 
+  public void commit_WriteSkewOnExistingRecordsWithSnapshot_ShouldProduceNonSerializableResult()
```

Comment:
This is just a new test verifying if write skew actually happens in SNAPSHOT. Just added for confirmation.